src/public/sharing.ts

```diff
@@ -0,0 +1,147 @@
+import { ensureInitialized } from '../internal/internalAPIs';
+import { sendMessageToParent } from '../internal/communication';
+import { ErrorCode, SdkError } from './interfaces';
+import { FrameContexts } from './constants';
+
+export namespace sharing {
+  export const SharingAPIMessages = {
+    shareWebContent: 'sharing.shareWebContent',
+  };
+
+  // More types can be added as we expand share capability
+  type ContentType = 'URL';
+
+  interface IBaseSharedContent {
+    type: ContentType;
+  }
+
+  // More types can be added as we expand share capability
+  export type IShareRequestContentType = IURLContent;
+
+  export interface IShareRequest<T> {
+    content: T[];
+  }
+
+  export interface IURLContent extends IBaseSharedContent {
+    type: 'URL';
+
+    /**
+     * Required URL
+     */
+    url: string;
+
+    /**
+     * Default initial message text
+     */
+    message?: string;
+
+    /**
+     * Show URL preview, defaults to true
+     */
+    preview?: boolean;
+  }
+
+  /**
+   * @private
+   * Feature is under development
+   *
+   * Opens a share dialog for web content
+   * @param shareWebContentRequest web content info
+   * @param callback optional callback
+   */
+  export function shareWebContent(
+    shareWebContentRequest: IShareRequest<IShareRequestContentType>,
+    callback?: (err?: SdkError) => void,
+  ): void {
+    if (!validateNonEmptyContent(shareWebContentRequest, callback)) {
+      return;
+    }
+
+    if (!validateTypeConsistency(shareWebContentRequest, callback)) {
+      return;
+    }
+
+    if (!validateContentForSupportedTypes(shareWebContentRequest, callback)) {
+      return;
+    }
+
+    ensureInitialized(
+      FrameContexts.content,
+      FrameContexts.sidePanel,
+      FrameContexts.task,
```

Comment:
Is task a valid frame context here? Are we going to support opening a share from from within a task module? That can be tricky since it would require opening one dialog from inside of another. Same comment applies to the stage.

Reply:
I think it's safe to ensure that these contexts are available just in case and let the Teams framework decide how to handle dialog on top of dialog scenarios.